tests/integration/targets/ndo_tenant_span_session/tasks/main.yml

```diff
@@ -0,0 +1,422 @@
+          - cm_update_span_session.current.sourceGroup.sources.1.epgName == nm_update_span_session.current.sourceGroup.sources.1.epgName == "ansible_test_epg_4"
+          - cm_update_span_session.current.sourceGroup.sources.1.epgSchemaName == nm_update_span_session.current.sourceGroup.sources.1.epgSchemaName == create_schema.current.displayName
+          - cm_update_span_session.current.sourceGroup.sources.1.epgSchemaId == nm_update_span_session.current.sourceGroup.sources.1.epgSchemaId == create_schema.current.id
+          - cm_update_span_session.current.sourceGroup.sources.1.epgTemplateName == nm_update_span_session.current.sourceGroup.sources.1.epgTemplateName == create_schema.current.templates.0.displayName
+          - cm_update_span_session.current.sourceGroup.sources.1.epgTemplateId == nm_update_span_session.current.sourceGroup.sources.1.epgTemplateId == create_schema.current.templates.0.templateID
+          - cm_update_span_session.current.uuid == nm_update_span_session.current.uuid
+          - nm_update_span_session_again is not changed
+          - nm_update_span_session_again.previous == nm_update_span_session_again.current == nm_update_span_session.current
+          - nm_update_span_session_name is changed
+          - nm_update_span_session_name.previous.name == "ansible_span_session"
+          - nm_update_span_session_name.current.name == "ansible_span_session_1"
+
+    # QUERY
+    - name: Create another SPAN Session
+      cisco.mso.ndo_tenant_span_session:
+        <<: *add_span_session
+        name: ansible_span_session_2
+      register: nm_span_session_2
+
+    - name: Query one SPAN Session by name
+      cisco.mso.ndo_tenant_span_session:
+        <<: *mso_info
+        name: ansible_span_session_2
+        template: ansible_monitoring_tenant_policy_template
+        state: query
+      register: query_name
+
+    - name: Query one SPAN Session by UUID
+      cisco.mso.ndo_tenant_span_session:
+        <<: *mso_info
+        uuid: '{{ nm_span_session.current.uuid }}'
+        template: ansible_monitoring_tenant_policy_template
+        state: query
+      register: query_uuid
+
+    - name: Query all SPAN Sessions
+      cisco.mso.ndo_tenant_span_session: &query_all_span_sessions
+        <<: *mso_info
+        template: ansible_monitoring_tenant_policy_template
+        state: query
+      register: query_all
+
+    - name: Assert SPAN Session is queried
+      ansible.builtin.assert:
+        that:
+          - query_name is not changed
+          - query_uuid is not changed
+          - query_all is not changed
+          - query_name.current == nm_span_session_2.current
+          - query_uuid.current == nm_update_span_session_name.current
+          - query_all.current | length == 2
+          - query_all.current[0].name == "ansible_span_session_1"
+          - query_all.current[1].name == "ansible_span_session_2"
+
+    # ERRORS TODO
```

Comment:
no errors? or still working on it? or Update SPAN Session with non-existing UUID the only error?

Reply:
No other module specific errors. Removed this note. 👍